src/datasources/synapse.datasource.ts

```diff
@@ -81,14 +81,15 @@ const config = {
     {
       template: {
         method: "GET",
-        url: baseURL + "/_matrix/client/r0/directory/room/{!name:string}",
+        url: baseURL + "/_synapse/admin/v1/rooms?search_term={!name:string}",
```

Comment:
Instead of changing the complete URL, I would check if there is the new version for the same endpoint that returns only one room.